analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -81,9 +81,9 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
                     var isExpression = (BinaryExpressionSyntax)c.Node;
-                    if (c.SemanticModel.GetTypeInfo(isExpression.Left).Type is { } objectToCast
+                    if (c.SemanticModel.GetTypeInfo(isExpression.Left).Type is var objectToCast
```

Comment:
`public ITypeSymbol Type { get; }` : `The type of the expression represented by the syntax node. For expressions that do not have a type, null is returned.`

the below throws
```
void Main()
{
	if (X() is var objectToCast)
	{
		Console.WriteLine(objectToCast.ToString());	
	}
	
	object X() => null;
}
```

Reply:
nevermind, the nullcheck is done inside `IsClass`
(which is good for extension methods however it's un-intuitive for the reader)